encryption.go

```diff
@@ -104,6 +109,26 @@ func DecryptMessage(ctx context.Context, id string, ct []byte, keySource X25519P
 		return fmt.Errorf("(%s) error deriving shared encryption key: %w", op, err)
 	}
 
+	err = decryptWithKey(ctx, id, ct, sharedKey, result)
+
+	// If decryption fails with the current key, try with the previous key, if present
+	if err != nil {
+		prevId, previousKey, prevErr := keySource.PreviousKey()
```

Comment:
We don't have to do this now but might be good to use a multierror here so we can return errors for both keys.